controller/src/controller/models/domain/rsptag.py

```diff
@@ -130,6 +163,22 @@ class RSPImageTag:
     display_name: str
     """Human-readable display name."""
 
+    date: datetime.datetime | None = None
+    """When the image was created, or as close as we can get to that.
+    We try to derive this from the tag string: For RSP daily or weekly
+    tags (or experimentals in one of those formats), we can calculate
+    this to within a day or a week, which is good enough for display
+    purposes.  Otherwise, we may be able to extract this info from
+    the registry, but even if we can, it may be image upload time
+    rather than creation time.
+    """
+
+    age: datetime.timedelta | None = None
+    """This is a calculated field, rather than a stored one.  This tells us
+    our best guess as to how old an image is and relies on the date field,
+    which may be unknown or inaccurate.
+    """
```

Comment:
Since this is a calculated field, it should be a property function or a method, not a data element. That will drastically simplify the code since then you don't have to calculate it in the constructor and you can drop `age` from most of the tests except for an explicit test of age. You can also get rid of the `recalculate_age` method.

A property is simplest, but it has the problem that the basis for calculating the age will keep changing since presumably it would use `datetime.now`. For menu purposes, I don't think you'll want that; you'll want to use the same basis for the whole menu calculation. So I would make this a method that takes a `datetime` argument to serve as the basis point for the age (which could be `None` to use now).